src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs

```diff
@@ -333,6 +335,16 @@ private static void SetPgAdminEnvironmentVariables(EnvironmentCallbackContext co
         // You need to define the PGADMIN_DEFAULT_EMAIL and PGADMIN_DEFAULT_PASSWORD or PGADMIN_DEFAULT_PASSWORD_FILE environment variables.
         context.EnvironmentVariables.Add("PGADMIN_DEFAULT_EMAIL", "admin@domain.com");
         context.EnvironmentVariables.Add("PGADMIN_DEFAULT_PASSWORD", "admin");
+
+        // When running in the context of Codespaces we need to set some additional environment
+        // varialbes so that PGAdmin will trust the forwarded headers that Codespaces port
+        // forwarding will send.
+        var codespaceOptions = context.ExecutionContext.ServiceProvider.GetRequiredService<IOptions<CodespacesOptions>>();
+        if (context.ExecutionContext.IsRunMode && codespaceOptions.Value.IsCodespace)
+        {
+            context.EnvironmentVariables["PGADMIN_CONFIG_PROXY_X_HOST_COUNT"] = "1";
+            context.EnvironmentVariables["PGADMIN_CONFIG_PROXY_X_PREFIX_COUNT"] = "1";
+        }
```

Comment:
Can we invert the dependency so that integrations don't need to worry about code spaces?

Reply:
The code that figures out what extra environment variables need to be injected into which resource needs to go somewhere, given the field of integrations is unbounded, and the specific environment variables tied to the integration it makes sense to do it this way I think.

Reply:
I think that is a possibility, but the characteristics of each proxy is probably going to have an impact on these variables anyway. So you might end up with a compound condition anyway, at which point it hasn't helped you.

I think that would be premature right now.